tests/test_modeling_utils.py

```diff
@@ -1171,6 +1172,151 @@ def test_safetensors_torch_from_torch_sharded(self):
             self.assertTrue(torch.equal(p1, p2))
 
 
+@require_torch
+class ModelOnTheFlyConversionTester(unittest.TestCase):
+    @classmethod
+    def setUpClass(cls):
+        cls.user = "huggingface-hub-ci"
+        cls.repo_name = f"{cls.user}/test-model-on-the-fly"
+        cls._token = os.getenv("HUGGINGFACE_PRODUCTION_USER_TOKEN", None)
+
+        if cls._token is None:
+            raise ValueError("Cannot run tests as secret isn't setup.")
+
+        HfFolder.save_token(cls._token)
+        cls.api = HfApi(token=cls._token)
+
+        if cls.api.repo_exists(cls.repo_name):
+            cls.api.delete_repo(cls.repo_name)
```

Comment:
Can be removed once the repo_id is made unique (this is the step that would break the tests if run in multiple CIs)